openedx/features/course_experience/views/course_outline.py

```diff
@@ -30,21 +32,75 @@ def render_to_fragment(self, request, course_id=None, page_context=None, **kwarg
         if not course_block_tree:
             return None
 
-        content_milestones = self.get_content_milestones(request, course_key)
+        # TODO: EDUCATOR-2283 Remove this check when the waffle flag is turned on in production
+        if waffle.new_course_outline_enabled(course_key=course_key):
+            xblock_display_names = self.create_xblock_id_and_name_dict(course_block_tree)
 
-        context = {
-            'csrf': csrf(request)['csrf_token'],
-            'course': course_overview,
-            'blocks': course_block_tree,
-            'gated_content': content_milestones
-        }
-        html = render_to_string('course_experience/course-outline-fragment.html', context)
-        return Fragment(html)
+            gated_content = self.get_content_milestones(request, course_key)
+
+            context = {
+                'csrf': csrf(request)['csrf_token'],
+                'course': course_overview,
+                'blocks': course_block_tree,
+                'gated_content': gated_content,
+                'xblock_display_names': xblock_display_names
+            }
+            # TODO: EDUCATOR-2283 Rename this file to course-outline-fragment.html
+            html = render_to_string('course_experience/course-outline-fragment-new.html', context)
+            return Fragment(html)
+
+        else:
+            content_milestones = self.get_content_milestones_old(request, course_key)
+
+            context = {
+                'csrf': csrf(request)['csrf_token'],
+                'course': course_overview,
+                'blocks': course_block_tree,
+                'gated_content': content_milestones
+            }
+            # TODO: EDUCATOR-2283 Remove this file
+            html = render_to_string('course_experience/course-outline-fragment-old.html', context)
+            return Fragment(html)
+
+    def create_xblock_id_and_name_dict(self, course_block_tree, xblock_display_names={}):
```

Comment:
Don't default `xblock_display_names` to an empty data structure - let it default to `None` instead, then have the function create a new dict if the parameter value is `None`.
https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html